src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -505,58 +523,75 @@ private void onMqttSubscribe(
                     }
                 }
 
-                for (int progress = offset; progress < limit; progress = subscription.limit())
+                for (int progress = offset, index = 0; progress < limit; progress = subscription.limit(), index++)
```

Comment:
Suggest creating an `MqttSubscription` outside the group and keeping it in a map keyed by `packetId`, then passing `packetId` and `ackIndex` to each `MqttServerStream` that is created.

For each non-routable server stream, explicitly set the failure reason code at the corresponding `ackIndex`.

If all server streams were non-routable, then `onMqttSubscribe` should respond immediately because all reason codes are known.

Otherwise, when each `MqttServerStream` gets an initial throttle frame:
  `RESET` should trigger `onSubscribeFailed(packetId, ackIndex)`, and
  `WINDOW` should trigger `onSubscribeSucceeded(packetId, ackIndex)`

`onSubscribeFailed` and `onSubscribeSucceeded` would behave similarly, choosing to report a failure or success `reasonCode` for the `ackIndex`, and detecting when all reason codes had reported in, triggering delivery of the `SUBACK` with complete list of reason codes back to the client.